### PR TITLE
Adds eye contact

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
I was walking down the street and made eye contact with someone and my brain thought to itself, "what if you could do that in ss13?" So long story short, examining someone who has examined you in the last second while within a 5 tile range will show the both of you a small message saying you guys looked at each other and made eye contact. If your face is obscured by a mask or headgear, the other person won't see the message. 

[](https://i.imgur.com/1VeJsjK.png)
As you can see, the message is smaller than usual so that it's unobstructive and easy to miss. Obviously this is just an example and catatonic people can't examine you.

### Niche cases
- Making eye contact with people while having the social anxiety quirk has a chance to make you jitter/stutter/freeze up for a moment along with a negative moodie. You also have a small chance to make eye contact with inanimate objects/brainless mobs because of your nerves.
[](https://i.imgur.com/A3THUTi.png)

- If an obsession target makes eye contact with their stalker and the stalker was the first to look, there's a 50% chance the target gets a message about the stalker staring at them, possibly blowing their cover. This can be sidestepped by hiding your face as a stalker.
[](https://i.imgur.com/SCJONll.png)

### Code improvement
Atoms will now properly be removed from recent_examine lists if qdel'd while in there

## Why It's Good For The Game
Realistically speaking, making eye contact with someone normally means nothing gameplay wise (obsession aside, that just fits well fluffwise), just that you both looked at each other within one second. It does, however, strike the perfect level of "wait why were they looking at me/is that important?" that catches people off guard and can prompt unexpected interactions. Imagining people getting flustered over being caught looking and wondering if they should say something sounds very funny to me.

Also this could probably have a neat interaction with social anxiety where making eye contact with someone can have a chance to give you a brief panic spike.

## Changelog
:cl: Ryll/Shaps
add: Examining someone who has examined you within the last second will give you both a small message saying you made eye contact. Having your face obscured by a helmet or mask will prevent the other person from noticing, no matter who looked first.
add: Those with social anxiety may find eye contact to be stressful, and those who foster unhealthy obsessions over other crewmembers may outright staredown their prey.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
make sure it gives you a bad moodlet if you have parasocial quirks

**Comment by infoman-cor** [comment]
What if we made eye contact in dorms 4 😳?

**Comment by Jared-Fogle** [comment]
[This isn't funny. You want to start making eye contact? I'm making eye contact with you. And I don't fuck around.](https://youtu.be/JaTYFs380rI?t=45)

**Comment by wesoda25** [comment]
I think it would be better if you implemented "sight" verbs, like bay does. ex: George Melons looks at you. Cuban Pete examines your ID.

**Comment by Ryll-Ryll** [comment]
> I think it would be better if you implemented "sight" verbs, like bay does. ex: George Melons looks at you. Cuban Pete examines your ID.
> 
> edit: oh this is more of a meme thing so nevermind.

It's not a meme, it's a small bit of fluff. I don't even know what "sight verbs" is supposed to mean.

**Comment by tralezab** [comment]
Make paranoid people always see people staring at them, this has a ton of potential

**Comment by Jared-Fogle** [comment]
> I think it would be better if you implemented "sight" verbs

If it takes any additional effort, nobody will use them. This is great because it'll happen just through normal gameplay.

**Comment by wesoda25** [comment]
> It's not a meme, it's a small bit of fluff. I don't even know what "add sight verbs" is supposed to mean.

Meme wasn't the right word, I meant just a sort of non serious small scale PR. Also I meant visible examine messages, such as the examples I provided.

**Comment by GuillaumePrata** [comment]
The part about making eye contact with the floor also trigger the jitter/stutter/freeze and negative mood?
Because picking the quirk to interact with players is fine, but getting these side effects because you find a new item and want to know if the description is funny or informative or want to examine a reinforced window to see the steps for deconstruction would be quite annoying.

**Comment by RaveRadbury** [comment]
@Ryll-Ryll bay gives feedback to observers when someone looks at someone else or examines something.

**Comment by Ryll-Ryll** [comment]
> The part about making eye contact with the floor also trigger the jitter/stutter/freeze and negative mood?
> Because picking the quirk to interact with players is fine, but getting these side effects because you find a new item and want to know if the description is funny or informative or want to examine a reinforced window to see the steps for deconstruction would be quite annoying.

nope, making eye contact with the floor just shows the text, you only chance the bad effects if you actually meet eyes with another person

**Comment by ATH1909** [comment]
yet another amazing ryll PR

> If an obsession target makes eye contact with their stalker and the stalker was the first to look, there's a 50% chance the target gets a message about the stalker staring at them, possibly blowing their cover. This can be sidestepped by hiding your face as a stalker.

These are, like, way too many conditions. Just make making eye contact with your stalker (regardless of who initiated it) give you the message 100% of the time (provided that their face isn't covered).

**Comment by KeRSedChaplain** [comment]
finally, now we can make endermen from minecraft

**Comment by Ryll-Ryll** [comment]
> These are, like, way too many conditions. Just make making eye contact with your stalker (regardless of who initiated it) give you the message 100% of the time (provided that their face isn't covered).

No

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -432,10 +434,41 @@
...
+		return
+
+	var/mob/living/carbon/examined_carbon = examined_mob
+	// check to see if their face is blocked (or if they're not a carbon, in which case they can't block their face anyway)
+	if(!istype(examined_carbon) || (!(examined_carbon.wear_mask && examined_carbon.wear_mask.flags_inv & HIDEFACE) && !(examined_carbon.head && examined_carbon.head.flags_inv & HIDEFACE)))
+		if(SEND_SIGNAL(src, COMSIG_MOB_EYECONTACT, examined_mob, TRUE) != COMSIG_BLOCK_EYECONTACT)
+			var/msg = "<span class='smallnotice'>You make eye contact with [examined_mob].</span>"
+			addtimer(CALLBACK(GLOBAL_PROC, .proc/to_chat, src, msg), 3) // so the examine signal has time to fire and this will print after
+
+	var/mob/living/carbon/us_as_carbon = src // i know >casting as subtype, but this isn't really an inheritable check
```
Comment (nemvar) [inline code/modules/mob/mob.dm:466]: Yes it is inheritable, just make another proc.

**Comment by OverlyGeneric** [comment]
finally, now medusa spells can work.

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (Ryll-Ryll) [reply code/modules/mob/mob.dm:466]: I'll add a face visible proc, but what's the issue with making eye contact with simple mobs? Lightgeists can just be handwaved as what appears to be eye contact even if it's unclear whether it has an eye, that's no reason to disable it for the entire class of mobs

**Comment by Mickyan** [comment]
shouldn't the visible message check for HIDEEYES instead of HIDEFACE?

**Comment by Ryll-Ryll** [comment]
Nah cause you can still usually tell if someone is looking at you with shades (also as anyone who's tried eyestabbing knows, literally everyone walks around wearing glasses which would ruin the point of this)

**Comment by PKPenguin321** [comment]
>Make paranoid people always see people staring at them, this has a ton of potential

This is neat and would be easy, paranoia is a genetics thing btw

**Comment by Mickyan** [comment]
okay but now balaclavas can be used to avoid eye contact because they have HIDEFACE and no HIDEEYES

**Pull request merged**